Lib/test/test_type_params.py

```diff
@@ -116,15 +116,14 @@ class ClassA[X]:
         self.assertIs(cls.__annotations__["X"], int)
 
     def test_name_non_collision_11(self):
```

Comment:
Is `non_collision_11` still the best name for this test?

If it's testing the lack of a name collision, then it probably shouldn't use `nonlocal` and raise an error. If it's testing that the use of nonlocal raises an error, then it should probably have a different name.

Reply:
Agree. I'll actually move this and the other `nonlocal` test into a new test class and add a few more cases.